python/desc/twinkles/twinklesCatalogDefs.py

```diff
@@ -63,7 +78,109 @@ def get_shorterFileNames(self):
     #    'shorterFileNames'
     column_outputs = ['prefix', 'uniqueId', 'raPhoSim', 'decPhoSim',
                       'phoSimMagNorm', 'shorterFileNames', 'redshift',
-                      'shear1', 'shear2', 'kappa', 'raOffset', 'decOffset',
+                      'gamma1', 'gamma2', 'kappa', 'raOffset', 'decOffset',
                       'spatialmodel', 'internalExtinctionModel',
                       'galacticExtinctionModel', 'galacticAv', 'galacticRv']
     cannot_be_null = ['x0', 't0', 'z', 'shorterFileNames']
+
+class TwinklesCompoundInstanceCatalog(CompoundInstanceCatalog):
+
+    def write_catalog(self, filename, chunk_size=None, write_header=True, write_mode='w'):
```

Comment:
How is this different from `CompoundInstanceCatalog.write_catalog()`?

Reply:
In line 179-180 it adds:
```
 +                compound_dbo.mjd = self._obs_metadata.mjd.TAI
 +                compound_dbo.specFileMap = twinkles_spec_map
```
which the sprinkler needs access to.

Reply:
Right. Now I remember talking about that.

Now that I see this solution in practice, it makes me uneasy. Could we just add a line to DESCQAObject.query_columns that sets self.obs_metadata = obs_metadata. This would give you access to MJD once the query is run. As far as the twinkles_spec_map, could that be handled at the InstanceCatalog level, the way that we handle the full file path mapping in "normal" PhoSim InstanceCatalogs

https://github.com/lsst/sims_catUtils/blob/master/python/lsst/sims/catUtils/exampleCatalogDefinitions/phoSimCatalogExamples.py#L220

?

Reply:
Or should we add a `query_metadata` dict to `DESCQAObject` that can contain `obs_metadata`?  That way it is clear(er) that this is data associated with a specific query, rather than data associated with the `DESCQAObject` in any permanent sense.